src/components/shared/Entity/Details/Spec/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { Grid, Stack, ToggleButtonGroup, Typography } from '@mui/material';
+import LiveSpecEditor from 'components/editor/LiveSpec';
+import ExternalLink from 'components/shared/ExternalLink';
+import OutlinedToggleButton from 'components/shared/OutlinedToggleButton';
+import { useEntityType } from 'context/EntityContext';
+import { outlinedToggleButtonGroupStyling } from 'context/Theme';
+import { useCallback, useMemo, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import CollectionSpecViews from './CollectionViews';
+
+export type SpecPresentation = 'table' | 'code';
+
+function Spec() {
+    const entityType = useEntityType();
+
+    const [presentation, setPresentation] = useState<SpecPresentation>('table');
+
+    const evaluatePresentation = useCallback(
+        (value: SpecPresentation, invertedValue: SpecPresentation): void => {
+            const evaluatedValue =
+                presentation === value ? invertedValue : value;
+
+            setPresentation(evaluatedValue);
+        },
+        [presentation, setPresentation]
+    );
+
+    const docsLink = useMemo(() => {
+        switch (entityType) {
+            case 'capture':
+                return 'https://docs.estuary.dev/concepts/captures/#specification';
+            case 'materialization':
+                return 'https://docs.estuary.dev/concepts/materialization/#specification';
+            default:
+                return 'https://docs.estuary.dev/concepts/collections/#specification';
+        }
+    }, [entityType]);
```

Comment:
One day we really should move some more stuff into a context for all the little differences between entity types

Reply:
Agreed, a similar thought has crossed my mind.